### PR TITLE
Adding new data access methods to IdentityDataStoreService

### PR DESCRIPTION
**Pull request by ZiyamSanthosh**
### Proposed changes in this pull request

[List all changes you want to add here. If you fixed an issue, please
add a reference to that issue as well.]

- This PR adds more methods to access data in IdentityDatsStore through IdentityDataStoreService

### When should this PR be merged

[Please describe any preconditions that need to be addressed before we
can merge this pull request.]

### Follow up actions

[List any possible follow-up actions here; for instance, testing data
migrations, software that we need to install on staging and production
environments.]

-

#### General

- [ ] **Is this PR explained thoroughly?** All code changes must be accounted for in the PR description.
- [ ] **Is the PR labeled correctly?**

#### Functionality

- [ ] **Are all requirements met?** Compare implemented functionality with the requirements specification.
- [ ] **Does the UI work as expected?** There should be no Javascript errors in the console; all resources should load. There should be no unexpected errors. Deliberately try to break the feature to find out if there are corner cases that are not handled.

#### Code

- [ ] **Do you fully understand the introduced changes to the code?** If not ask for clarification, it might uncover ways to solve a problem in a more elegant and efficient way.
- [ ] **Does the PR introduce any inefficient database requests?** Use the debug server to check for duplicate requests.
- [ ] **Are all necessary strings marked for translation?** All strings that are exposed to users via the UI must be [marked for translation](https://docs.djangoproject.com/en/1.10/topics/i18n/translation/).

#### Tests

- [ ] **Are there sufficient test cases?** Ensure that all components are tested individually; models, forms, and serializers should be tested in isolation even if a test for a view covers these components.
- [ ] **If this is a bug fix, are tests for the issue in place?**  There must be a test case for the bug to ensure the issue won’t regress. Make sure that the tests break without the new code to fix the issue.
- [ ] **If this is a new feature or a significant change to an existing feature?** has the manual testing spreadsheet been updated with instructions for manual testing?

#### Security

- [ ] **Confirm this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets.**
- [ ] **Are all UI and API inputs run through forms or serializers?**
- [ ] **Are all external inputs validated and sanitized appropriately?**
- [ ] **Does all branching logic have a default case?**
- [ ] **Does this solution handle outliers and edge cases gracefully?**
- [ ] **Are all external communications secured and restricted to SSL?**

#### Documentation

- [ ] **Are changes to the UI documented in the platform docs?** If this PR introduces new platform site functionality or changes existing ones, the changes should be documented.
- [ ] **Are changes to the API documented in the API docs?** If this PR introduces new API functionality or changes existing ones, the changes must be documented.
- [ ] **Are reusable components documented?** If this PR introduces components that are relevant to other developers (for instance a mixin for a view or a generic form) they should be documented in the Wiki.

### CODE REVIEW

File: components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/store/JDBCIdentityDataStore.java [hunk 1]
```diff
@@ -559,6 +611,14 @@ private static class SQLQuery {
                 "FROM IDN_IDENTITY_USER_DATA " +
                 "WHERE DATA_KEY = ? AND DATA_VALUE LIKE ? AND TENANT_ID = ? AND USER_NAME LIKE ?";
 
+        public static final String FILTER_USERS_BY_DATA_KEY_LESS_THAN_DATA_VALUE =
+                "SELECT USER_NAME, DATA_VALUE FROM IDN_IDENTITY_USER_DATA WHERE " +
+                        "DATA_KEY = ? AND TENANT_ID = ? AND DATA_VALUE < ?";
+
+        public static final String FILTER_USERS_BY_DATA_KEY_LESS_THAN_AND_GREATER_THAN_DATA_VALUES =
+                "SELECT USER_NAME, DATA_VALUE FROM IDN_IDENTITY_USER_DATA WHERE " +
+                        "DATA_KEY = ? AND TENANT_ID = ? AND DATA_VALUE < ? AND DATA_VALUE > ?";
```
Comment (indeewari) [inline components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/store/JDBCIdentityDataStore.java:620]: Please double-check the order of the parameters. The start and end boundaries should be, 

startValue < DATA_VALUE < endValue       -->  
DATA_VALUE > startValue and DATA_VALUE < endValue    -->  
DATA_VALUE > ? and DATA_VALUE < ?

What I see here is

DATA_VALUE < startValue and DATA_VALUE > endValue  -->  endValue < DATA_VALUE < startValue

File: components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/store/JDBCIdentityDataStore.java [hunk 2]
```diff
@@ -387,6 +387,58 @@ public List<String> listPaginatedUsersNames(List<ExpressionCondition> identityCl
...
+    public  List<String> getUserNamesBetweenProvidedClaimValues(String claimURI, String startValue, String endValue,
+                                                                int tenantId) throws IdentityException {
+
+        String sqlStmt = SQLQuery.FILTER_USERS_BY_DATA_KEY_LESS_THAN_AND_GREATER_THAN_DATA_VALUES;
+        List<String> userNames = new ArrayList<>();
+        try (Connection connection = IdentityDatabaseUtil.getDBConnection(true)) {
+            try (PreparedStatement prepStmt = connection.prepareStatement(sqlStmt)) {
+                prepStmt.setString(1, claimURI);
+                prepStmt.setInt(2, tenantId);
+                prepStmt.setString(3, startValue);
```
Comment (indeewari) [inline components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/store/JDBCIdentityDataStore.java:425]: Please refer my comment on the sql query

File: components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/store/JDBCIdentityDataStore.java (hunk 1 above)
Comment (ZiyamSanthosh) [reply components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/store/JDBCIdentityDataStore.java:620]: Fixed with https://github.com/wso2-extensions/identity-governance/pull/726/commits/41e2a3f5a5f65efc1f3d48b4fdece020ae3ece6b

File: components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/store/JDBCIdentityDataStore.java (hunk 2 above)
Comment (ZiyamSanthosh) [reply components/org.wso2.carbon.identity.governance/src/main/java/org/wso2/carbon/identity/governance/store/JDBCIdentityDataStore.java]: Fixed with https://github.com/wso2-extensions/identity-governance/pull/726/commits/41e2a3f5a5f65efc1f3d48b4fdece020ae3ece6b

**Review by indeewari — APPROVED** [review]

**Pull request merged**